packages/vite/src/node/utils.ts

```diff
@@ -1246,3 +1262,8 @@ export function evalValue<T = any>(rawValue: string): T {
   `)
   return fn()
 }
+
+const escapeForRegexRE = /[-/\\^$*+?.()|[\]{}]/g
+export function escapeForRegex(str: string): string {
```

Comment:
nit: because I've also refactored this in another branch, I think `escapeRegex` is more succinct 😬 because this function itself escapes the regex syntax